### PR TITLE
Performance Optimization - Issue #37

### PR DESCRIPTION
**Pull request by TheColorRed**
Optimization changes to load files even if they error. and helps with issue that would freeze the editor.

#37 Performance Optimization

### CODE REVIEW

**Comment by TheColorRed** [comment]
@nevadascout Here is the new pull request without the 3rd party libraries.

**Comment by TheColorRed** [comment]
Found and fixed the Issue for Issue #47. There was a missing check to see if the element was null or not before calling `ProcessBranch`

**Comment by TheColorRed** [comment]
@nevadascout I have start a second phase of optimization. Once this pull gets pulled in I can show you phase two which loads large projects in under a second.

File: server/package.json [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+    "vscode-languageserver": "^1.1.0"
+  },
+  "devDependencies": {
+    "typescript": "^1.6.2"
+  },
+  "scripts": {
+    "compile": "installServerIntoExtension ../client ./package.json ./tsconfig.json && tsc -p .",
+    "watch": "installServerIntoExtension ../client ./package.json ./tsconfig.json && tsc --watch -p ."
+  }
+}
```
Comment (nevadascout) [inline server/package.json]: Can you set the indentation in this file back to 4 spaces instead of 2?

Comment (TheColorRed) [inline server/package.json]: Sure, I think npm did that with `npm install filequeue --save`

File: server/src/server.ts [hunk 2]
```diff
@@ -230,7 +234,7 @@ function addStaticClassMembers(toReturn: CompletionItem[], item:ClassNode)
             });
 
             if (!found) {
-                toReturn.push({ label: subNode.name, kind: CompletionItemKind.Method, detail: "method (static)", insertText: subNode.name + "()" });
+                toReturn.push({ label: subNode.name, kind: CompletionItemKind.Method, detail: "method (static)", insertText: subNode.name + "(" + stringifyMethodParams(subNode) +")" });
```
Comment (nevadascout) [inline server/src/server.ts]: Can you remove this from this pull request please? We'll discuss this separately in #44

File: server/src/server.ts [hunk 3]
```diff
@@ -307,7 +311,7 @@ function addClassPropertiesMethodsParentClassesAndTraits(toReturn: CompletionIte
 
     classNode.methods.forEach((subNode) => {
         var accessModifier = "method " + buildAccessModifier(subNode.accessModifier);
-        var insertText = subNode.name + "()";
+        var insertText = subNode.name + "(" + stringifyMethodParams(subNode) +")";
```
Comment (nevadascout) [inline server/src/server.ts]: As above - can you remove this from this pull request please? We'll discuss this separately in #44

File: server/src/server.ts [hunk 4]
```diff
@@ -342,6 +346,16 @@ function addClassPropertiesMethodsParentClassesAndTraits(toReturn: CompletionIte
...
+function stringifyMethodParams(subNode): string
+{
+    var params = [];
+    subNode.params.forEach(param => {
+        if(param.optional){return;}
+        params.push(param.name);
+    });
+    return params.join(', ');
+}
+
```
Comment (nevadascout) [inline server/src/server.ts]: As above

**Comment by nevadascout** [comment]
Made a few comments on minor things, other than that this looks good - thanks!

Once those changes are made I'll merge this in.

**Comment by TheColorRed** [comment]
@nevadascout Alright I have made the changes.

**Comment by nevadascout** [comment]
@TheColorRed Please add the extra performance info on #37 so we can keep track of it

**Pull request merged**